fish-rust/src/threads.rs

```diff
@@ -0,0 +1,72 @@
+//! The rusty version of iothreads from the cpp code, to be consumed by native rust code. This isn't
+//! ported directly from the cpp code so we can use rust threads instead of using pthreads.
+
+use crate::flog::FLOG;
+
+/// The rusty version of `iothreads::make_detached_pthread()`. We will probably need a
+/// `spawn_scoped` version of the same to handle some more advanced borrow cases safely, and maybe
+/// an unsafe version that doesn't do any lifetime checking akin to
+/// `spawn_unchecked()`[std::thread::Builder::spawn_unchecked], which is a nightly-only feature.
+///
+/// Returns a boolean indicating whether or not the thread was successfully launched. Failure here
+/// is not dependent on the passed callback and implies a system error (likely insufficient
+/// resources).
+pub fn spawn<F: FnOnce() + Send + 'static>(callback: F) -> bool {
```

Comment:
Hmm why not call this file `iothread.rs`, is it because we use qualified names? Fine I guess.

Reply:
Well it’s not specific to io threads. We never want to launch a new thread that is using our current signal mask. I don’t much like the generic thread name but I didn’t want to waste too much time bike-shedding. This damn PR took way longer and way more work than I was expecting!